plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp

```diff
@@ -293,6 +295,116 @@ class read_only {
       }
    }
 
+   static uint64_t get_table_index_name(const read_only::get_table_rows_params& p) {
+      // see multi_index packing of index name
+      const uint64_t table = p.table;
+      uint64_t index = table & 0xFFFFFFFFFFFFFFF0ULL;
+      EOS_ASSERT( index == table, chain::contract_table_query_exception, "Unsupported table name: ${n}", ("n", p.table) );
+
+      uint64_t pos = 0;
+      if (p.index_position.empty() || p.index_position == "first" || p.index_position == "primary") {
+      } else if (p.index_position == "second" || p.index_position == "secondary") {
+      } else if (p.index_position == "third" || p.index_position == "tertiary" || p.index_position == "ternary") {
+         pos = 1;
+      } else if (p.index_position == "fourth" || p.index_position == "quaternary") {
+         pos = 2;
+      } else if (p.index_position == "fifth" || p.index_position == "quinary") {
+         pos = 3;
+      } else if (p.index_position == "sixth" || p.index_position == "senary") {
+         pos = 4;
+      } else if (p.index_position == "seventh" || p.index_position == "septenary") {
+         pos = 5;
+      } else if (p.index_position == "eighth" || p.index_position == "octonary") {
+         pos = 6;
+      } else if (p.index_position == "ninth" || p.index_position == "nonary") {
+         pos = 7;
+      } else if (p.index_position == "tenth" || p.index_position == "denary") {
+         pos = 8;
+      } else {
+         try {
+            pos = fc::to_uint64( p.index_position );
+         } catch(...) {
+            EOS_ASSERT( false, chain::contract_table_query_exception, "Invalid index_position: ${p}", ("p", p.index_position));
+         }
+         if (pos < 2) {
+            pos = 0;
+         } else {
+            pos -= 2;
+         }
+      }
+      index |= (pos & 0x000000000000000FULL);
+      return index;
+   }
+
+   template <typename IndexType, typename Scope, typename SecKeyType, typename ConvFn>
```

Comment:
It looks like `Scope` is no longer really needed as a template parameter and a relic of the days when we had multiple intrusive indices.  All of the callers use `by_secondary` and it is only used in a way where `by_secondary` is always the right choice.  We should probably simplify it out.